packages/flutter/lib/src/material/pickers/calendar_date_range_picker.dart

```diff
@@ -543,10 +543,9 @@ class _MonthItem extends StatelessWidget {
     // formatted full date.
     String semanticLabel = '${localizations.formatDecimal(day)}, ${localizations.formatFullDate(dayToBuild)}';
     if (isSelectedDayStart) {
-      // TODO(darrenaustin): localize 'Start Date' and 'End Date'
-      semanticLabel = 'Start Date ' + semanticLabel;
+      semanticLabel = '${localizations.dateRangeStartLabel} $semanticLabel';
```

Comment:
Do we know that for non-English messages, this order of the strings would be desirable? (ie. `Start Date` comes before `$semanticLabel`). It might be important to share some context on words/strings regarding the label (what might come before/after the string), as well as whether it will be read aloud, visually read, or both (this came up as a question recently with the Cupertino tabs semantics label).

Reply:
Good point. I just changed these two labels to parameterized messages, so they can adjust the order as they like.